components/compliance-service/inspec-agent/resolver/resolver.go

```diff
@@ -91,6 +90,60 @@ func (r *Resolver) ResolveJob(ctx context.Context, job *jobs.Job) ([]*types.Insp
 	r.scannerServer.UpdateJobNodeCount(job.Id, len(nodeJobs))
 	return nodeJobs, nil
 }
+
+type nodeInfo struct {
+	UUID              string
+	CloudID           string
+	Name              string
+	Environment       string
+	CloudAccountID    string
+	ManagerID         string
+	SSM               bool
+	MachineIdentifier string
+	Tags              []*common.Kv
+}
+
+func assembleJob(job *jobs.Job, node nodeInfo, secrets []*inspec.Secrets, tc inspec.TargetBaseConfig) *types.InspecJob {
+	if len(node.UUID) == 0 {
+		id, err := uuid.NewV4()
+		if err != nil {
+			logrus.Warnf("unable to generate uuid for node")
+		}
+		node.UUID = id.String()
+	}
+
+	fullTc := inspec.TargetConfig{
+		TargetBaseConfig: tc,
+	}
+	if len(secrets) == 1 {
+		fullTc.Secrets = *secrets[0]
+	} else {
+		fullTc.SecretsArr = secrets
+	}
+
+	return &types.InspecJob{
+		InspecBaseJob: types.InspecBaseJob{
+			JobID:    job.Id,
+			JobName:  job.Name,
+			NodeEnv:  node.Environment,
+			NodeName: node.Name,
+			JobType:  job.Type,
+			Status:   job.Status,
+			NodeID:   node.UUID,
+		},
+		SSM:               node.SSM,
+		Timeout:           job.Timeout,
+		Retries:           job.Retries,
+		TargetConfig:      fullTc,
+		Profiles:          job.Profiles,
+		SourceID:          node.CloudID,
+		SourceAccountID:   node.CloudAccountID,
+		ManagerID:         node.ManagerID,
+		MachineIdentifier: node.MachineIdentifier,
```

Comment:
would be good to add a comment for what some of the non-obvious fields are/why they are used

Reply:
done on the struct definition